lib/intent-models/message.js

```diff
@@ -0,0 +1,123 @@
+
+module.exports = {}; 
+const IntentBase = require('./base');
+// eslint-disable-next-line no-unused-vars
+const Schema = require('../models/schema');
+const { IntentOperation } = require('./operation');
+const { IntentChannel } = require('./channel');
+const { IntentCorrelationId } = require('./correlation-id');
+const { IntentTag } = require('./tag');
+
+/**
+ * @class
+ * @alias module:@asyncapi/parser#IntentMessage
+ * @extends IntentBase
+ */
+class IntentMessage extends IntentBase {
+  /**
+   * @returns {string}
+   */
+  uid() { return 'test-uid'; }
+  /**
+   * @returns {Schema}
+   */
+  payload() { return new Schema({type: 'integer'}); }
+
+  /**
+   * @returns {IntentChannel[]}
+   */
+  channels() { return [new IntentChannel()]; }
+
+  /**
+   * @returns {IntentOperation[]}
+   */
+  operations() { return [new IntentOperation()]; }
+
+  /**
+    * @returns {boolean}
+    */
+  hasName() { return true; }
+  /**
+   * @returns {string|undefined}
+   */
+  name() { return 'some test name'; }
+
+  /**
+    * @returns {boolean}
+    */
+  hasSummary() { return true; }
+  /**
+    * @returns {string|undefined}
+    */
+  summary() { return 'some message summary'; }
+  
+  /**
+    * @returns {boolean}
+    */
+  hasTitle() { return true; }
+  /**
+    * @returns {string|undefined}
+    */
+  title() { return 'some message title'; }
+ 
+  /**
+    * @returns {boolean}
+    */
+  hasHeaders() { return true; }
+  /**
+    * @returns {Schema|undefined}
+    */
+  headers() { return new Schema({});}
+
+  /**
+   * 
+   * @param {string} bindingProtocol 
+   * @returns {boolean}
+   */
+  hasBinding(bindingProtocol) { return true; }
+  /**
+    * 
+    * @param {string} bindingProtocol 
+    * @returns {any|undefined} 
+    */
+  binding(bindingProtocol) {return {};}
+
+  /**
+   * 
+   * @returns {string} 
+   */
+  contentType() { return 'application/json';}
+  /**
+   * @returns {Object<string, any>[]}
+   */
+  examples() { return {}; }
```

Comment:
I feel like the return type can be better explained, especially since there is quite the confusion about what this example property may contain. What do you think?

Reply:
The correct return type is `[Map[string, any]]` where `string` is either `headers` or `payload`. 

But we also have to be careful about this, cause there is an incoming breaking change here: https://github.com/asyncapi/spec/issues/329 
Not sure if we can reduce the impact in any way 🤔

Reply:
As far as I see, https://github.com/asyncapi/spec/issues/329 will introduce a breaking change on the spec. 
As the return type will change on the Parser API, we could consider introducing right now a new model called `Example`. However, I don't see how we could make them handle the raw version (`[Map[string, any]]`) without adding a method that returns it explicitly. For me this sounds like a patch and not very intuitive.
We could also return either a `[Map[string, any]]` or an `Example` Object depending on the content of the examples json, however I wouldn't advocate for that as it's completely a bad practice, also not pretty suitable and common in strong typed languages.

I would just carry on with a possible breaking change here...

Reply:
Btw, isn't the type nomenclature `Map[string, any][]` rather than `[Map[string, any]]`?

Reply:
🤔 Yea I think it is 🧐

Reply:
I have changed the return type to `Map[string, any][]` in the mock